src/plonetheme/tokyo/browser/overrides/plone_app_z3cform/plone.app.z3cform.templates.macros.pt

```diff
@@ -105,24 +114,21 @@
             </tal:block>
 
           </metal:define>
-        </metal:fields-slot>
 
-        <metal:block define-slot="belowfields" />
+        </metal:fields-slot>
 
-        <metal:actions-slot define-slot="actions">
+        <metal:actions define-slot="actions">
           <metal:define define-macro="actions">
-            <div class="form-group" tal:condition="view/actions/values|nothing">
-              <tal:block repeat="action view/actions/values">
-                <input class="btn btn-primary" type="submit" tal:replace="structure action/render" />
-              </tal:block>
-            </div>
+            <tal:block repeat="action view/actions/values">
+              <tal:define tal:define="data action/json_data; btn_class python:'danger' if 'cancel' in action.name else 'primary'">
+                <input class="btn btn-${btn_class}" type="${data/type}" name="${data/name}" value="${action/value}" />
+              </tal:define>
+            </tal:block>
           </metal:define>
-        </metal:actions-slot>
+        </metal:actions>
 
         <tal:block tal:condition="view/enableCSRFProtection|nothing" tal:replace="structure context/@@authenticator/authenticator" />
 
-        <metal:block define-slot="formbottom" />
-
```

Comment:
Not sure what this was implemented for... Are you sure we don't need it anymore?